tests/scripts/deploy-validate-vault.sh

```diff
@@ -54,21 +54,44 @@ EOF
   openssl req -new -key "${TMPDIR}"/vault.key -subj "/CN=${SERVICE}.${NAMESPACE}.svc" -out "${TMPDIR}"/server.csr -config "${TMPDIR}"/csr.conf
 
   export CSR_NAME=vault-csr
+  
+  # Minimum 1.19.0 kubernetes version is required for certificates.k8s.io/v1 version
+  SERVER_VERSION=$(kubectl version --short | awk -F  "."  '/Server Version/ {print $2}')
+  MINIMUM_VERSION=19
+  if [ ${SERVER_VERSION} -lt ${MINIMUM_VERSION} ]
+  then
+    'cat <<EOF >"${TMPDIR}"/csr.yaml
+  apiVersion: certificates.k8s.io/v1beta1
+  kind: CertificateSigningRequest
+  metadata:
+    name: ${CSR_NAME}
+  spec:
+    groups:
+    - system:authenticated
+    request: $(cat ${TMPDIR}/server.csr | base64 | tr -d '\n')
+    usages:
+    - digital signature
+    - key encipherment
+    - client auth
+  EOF'
+  else
+    'cat <<EOF >"${TMPDIR}"/csr.yaml
+  apiVersion: certificates.k8s.io/v1
+  kind: CertificateSigningRequest
+  metadata:
+    name: ${CSR_NAME}
+  spec:
+    signerName: kubernetes.io/kube-apiserver-client  
```

Comment:
Using `kubernetes.io/kubelet-serving` would have been a good match but the permitted subjects - organizations are exactly ["system:nodes"], common name starts with "system:node:" so this won't work since vault uses `system:authenticated`.

Let's try with `kubernetes.io/legacy-unknown` for more flexibility.

Reply:
I have tried using `kubernetes.io/legacy-unknown` but CI is throwing error `The CertificateSigningRequest "vault-csr" is invalid: spec.signerName: Invalid value: "kubernetes.io/legacy-unknown": the legacy signerName is not allowed via this API version`

it's because of [signers](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#kubernetes-signers): `the stable CertificateSigningRequest API (version certificates.k8s.io/v1 and later) does not allow to set the signerName as kubernetes.io/legacy-unknown.`

Reply:
try `kubernetes.io/kube-apiserver-client`, when we were using certificates for admission controller, I used this and it was working.

Reply:
Okay